File: HotelBookingSystem/src/Shared/HotelBooking.Models/Entities/Room.cs
using System;

namespace HotelBooking.Models.Entities
{
    public class Room
    {
        public Guid RoomId { get; set; } = Guid.NewGuid();
        public string RoomNumber { get; set; } = string.Empty;
        public string RoomType { get; set; } = string.Empty;
        public int MaxOccupancy { get; set; }
        public decimal PricePerNight { get; set; }
        public string? Description { get; set; }
        public string? Amenities { get; set; } // JSON format
        public bool IsActive { get; set; } = true;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: HotelBookingSystem/hotel-booking-frontend/src/types/room.ts
export interface Room {
  roomId: string;
  roomNumber: string;
  roomType: string;
  maxOccupancy: number;
  pricePerNight: number;
  description: string;
  amenities: string[];
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface CreateRoomDto {
  roomNumber: string;
  roomType: string;
  maxOccupancy: number;
  pricePerNight: number;
  description: string;
  amenities: string[];
  isActive?: boolean;
}

export interface RoomAvailabilityQuery {
  checkInDate: string;
  checkOutDate: string;
  numberOfGuests: number;
}

export interface RoomAvailability {
  roomId: string;
  roomNumber: string;
  roomType: string;
  pricePerNight: number;
  isAvailable: boolean;
  totalPrice: number;
}
